contracts/src/Assets.sol

```diff
@@ -145,19 +149,77 @@ library Assets {
     ) internal returns (Ticket memory ticket) {
         AssetsStorage.Layout storage $ = AssetsStorage.layout();
 
-        // Lock the funds into AssetHub's agent contract
-        if (token != address(0)) {
-            // ERC20
-            _transferToAgent($.assetHubAgent, token, sender, amount);
-            ticket.value = 0;
+        // Lock Native Ether into AssetHub's agent contract
+        if (msg.value < amount) {
+            revert InvalidAmount();
+        }
+        payable($.assetHubAgent).safeNativeTransfer(amount);
+        ticket.value = amount;
+
+        ticket.dest = $.assetHubParaID;
```

Comment:
It seems that both `_sendEther` and `_sendNativeToken` share this ticket/payload creation code. Can we not factor that out?

Reply:
addressed in 0921ff9777a86e0be660bd46d36b6cef965dc171